packages/gatsby/cache-dir/head/index.js

```diff
@@ -0,0 +1,38 @@
+import React from "react"
+import { Helmet } from "react-helmet-async"
+import { useLocation } from "@reach/router"
+
+export const Head = props => {
+  const location = useLocation()
+
+  const { title, description, image } = props
+
+  // NOTE(@mxstbr): This code has many ternaries for the same things because
+  // react-helmet doesn't support fragments yet
+  // @see github.com/nfl/react-helmet/issues/342
+  return (
+    <Helmet>
+      {/* Title */}
+      {title && <title>{title}</title>}
+      {title && <meta name="twitter:title" content={title} />}
+      {title && <meta property="og:title" content={title} />}
+
+      {/* Description */}
+      {description && <meta name="description" content={description} />}
+      {description && <meta name="twitter:description" content={description} />}
+      {description && <meta property="og:description" content={description} />}
+
+      {/* Image */}
+      {image && <meta name="twitter:image" content={image} />}
```

Comment:
I don't believe this will work unless it's a full path, which requires the site URL to be pulled from somewhere.

Reply:
100%, this is something we can add warnings for later! Great idea 👍

Reply:
It's a bigger issue than that though. Right now, we're taking a string or the src that comes with image.  That's a relative path, not an absolute one. Even if we grab an absolute path, we need to know the domain the user will set in production, so we need some way to grab metadata. Ideally from gatsby-config probably.